### PR TITLE
2 setup devenv from scratch

### PR DESCRIPTION
**Pull request by matteodelucchi**
The basic structure for the new development environment towards abn 4.0 within a virtual environment and Rcpp support. Closing #2

### CODE REVIEW

**Comment by matteodelucchi** [comment]
Helpful resource for the future regarding C/C++ integration in R with `Rcpp`: https://www.rcpp.org/

**Review by j-i-l — COMMENTED** [review]
This includes changes introduced by `Rcpp.package.skeleton('abn2')`, right?

File: DESCRIPTION [hunk 1]
```diff
@@ -41,30 +11,8 @@ Author:
...
- 'abn' provides routines to help determine optimal Bayesian network models for a given data set, where these models are used to identify statistical dependencies in messy, complex data. The additive formulation of these models is equivalent to multivariate generalised linear modelling (including mixed models with iid random effects). 
- The usual term to describe this model selection process is structure discovery. 
- The core functionality is concerned with model selection - determining the most robust empirical model of data from interdependent variables. Laplace approximations are used to estimate goodness of fit metrics and model parameters, and wrappers are also included to the INLA package which can be obtained from <https://www.r-inla.org>. 
- The computing library JAGS <https://mcmc-jags.sourceforge.io> is used to simulate 'abn'-like data. 
- A comprehensive set of documented case studies, numerical accuracy/quality assurance exercises, and additional documentation are available from the 'abn' website <http://r-bayesian-networks.org>.
```
Comment (j-i-l) [inline DESCRIPTION:60]: This likely won't change much. Maybe we can keep it too?

Comment (matteodelucchi) [reply DESCRIPTION:60]: agree. Included the old description.

File: NAMESPACE [hunk 2]
```diff
@@ -1,140 +1,3 @@
...
-export(searchHeuristic)
-export(searchHillClimber)
-export(simulateAbn)
-export(simulateDag)
-export(skewness)
-export(std.area.under.grid)
-export(strsplits)
-export(tidy.cache)
-export(toGraphviz)
+useDynLib(abn2, .registration=TRUE)
```
[the remark this answers is not in the document]
Comment (matteodelucchi) [reply NAMESPACE]: Remnant from a workaround. Fixed it.

File: ChangeLog
[the remark this answers is not in the document]
Comment (matteodelucchi) [reply ChangeLog:1]: Thanks! Definitely we need a change log. The old `ChangeLog` is not in the desired format of `usethis` and CRAN requests the file called `NEWS.md`. I changed that accordingly.

**Pull request merged**